### ISSUE TITLE
My testing results is 63.039/75.899.Is it right?

### ISSUE THREAD

**Issue by wangkai602058765** [Ghostish/BAT#5]
Hello,i test BAT with the bat_kitti_car.ckpt you provided,My result is 63.039/75.899,Is it right?
![屏幕截图 2021-09-01 083952](https://user-images.githubusercontent.com/42930372/131593730-746e8f35-a21e-4adf-bfeb-e11ec856db24.png)

**Comment by Ghostish** [comment]
Hi, it is weird. I evaluated the uploaded checkpoint and got the reported results. 
![image](https://user-images.githubusercontent.com/7406087/131605515-697d10af-2ba2-46f4-8d49-bc081f40e5d8.png)

Could you provide some information about your running platform (e.g. version of the GPU, pytorch and CUDA)? 
How about retraining the model from the scratch, what do you get?

**Comment by wangkai602058765** [comment]
@Ghostish  I use 3080Ti,pytorch1.8.0,cuda11.1 and get the result 63.039/75.899. And then,i test the code on another computer(2080Ti,pytorch1.5.0,cuda10.2) and get the result 65.371/78.879.I dont't know why.Could you please give your opinion?thank you!!

**Comment by Ghostish** [comment]
> @Ghostish I use 3080Ti,pytorch1.8.0,cuda11.1 and get the result 63.039/75.899. And then,i test the code on another computer(2080Ti,pytorch1.5.0,cuda10.2) and get the result 65.371/78.879.I dont't know why.Could you please give your opinion?thank you!!

I suppose this may be due to the `random number generator (rng)` that we use to sample a fixed number of points in the function `regularize_pc()`. Though I fixed the seed during testing,  it may behave differently across different platforms.  

You can print out the `new_pts_idx` for the same object on different machines and check whether they are the same.

How about retraining the model from scratch using the  3080Ti ? Can you reproduce the results?

**Comment by wangkai602058765** [comment]
> > @Ghostish I use 3080Ti,pytorch1.8.0,cuda11.1 and get the result 63.039/75.899. And then,i test the code on another computer(2080Ti,pytorch1.5.0,cuda10.2) and get the result 65.371/78.879.I dont't know why.Could you please give your opinion?thank you!!
> 
> I suppose this may be due to the `random number generator (rng)` that we use to sample a fixed number of points in the function `regularize_pc()`. Though I fixed the seed during testing, it may behave differently across different platforms.
> 
> You can print out the `new_pts_idx` for the same object on different machines and check whether they are the same.
> 
> How about retraining the model from scratch using the 3080Ti ? Can you reproduce the results?

Ok，i will retrain the code  and tell you the result,thank you for your reply

**Comment by wangkai602058765** [comment]
> > @Ghostish I use 3080Ti,pytorch1.8.0,cuda11.1 and get the result 63.039/75.899. And then,i test the code on another computer(2080Ti,pytorch1.5.0,cuda10.2) and get the result 65.371/78.879.I dont't know why.Could you please give your opinion?thank you!!
> 
> I suppose this may be due to the `random number generator (rng)` that we use to sample a fixed number of points in the function `regularize_pc()`. Though I fixed the seed during testing, it may behave differently across different platforms.
> 
> You can print out the `new_pts_idx` for the same object on different machines and check whether they are the same.
> 
> How about retraining the model from scratch using the 3080Ti ? Can you reproduce the results?
I retrain the code tonight using 3080ti and cuda11.1.My best result is 63.84/75.61.

![屏幕截图 2021-09-02 165954](https://user-images.githubusercontent.com/42930372/131816173-9692ea41-37b2-4857-9300-554da71f686e.png)

**Comment by handworker** [comment]
Hi, 
Maybe I can provide another validation by using the provided pre-trained bat_kitti_cat.cpkt model:

--------------------------------------------------------------------------------
DATALOADER:0 VALIDATE RESULTS
{'precision/test': 76.09471893310547,
 'precision/test_epoch': 76.09471893310547,
 'success/test': 63.18337321281433,
 'success/test_epoch': 63.18337321281433}
--------------------------------------------------------------------------------

Environment: Cuda-11.0; PyTorch: 1.7.0; RTX 3090;

It seems that the success result is better and the precision is slightly inferior to the original result reported in Table 1 of the paper. Is it correct?
